src/app/shared/modals/other-user-profile/other-user-profile.component.ts

```diff
@@ -19,7 +19,7 @@ export class OtherUserProfileComponent {
 
   constructor(@Inject(MAT_DIALOG_DATA) public data:
     { userPic: string, userFirstName: string, userLastName: string, userXP: number },
-    public readonly matDialog: MatDialog) { }
+              public readonly matDialog: MatDialog) { }
```

Comment:
you do not use `matDialog` in the template so make in private

Reply:
fixed